src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -268,27 +284,55 @@ def deregister_stack(self, name: str) -> None:
 
     # Private interface implementations:
 
-    def _create_stack(
-        self, name: str, stack_configuration: Dict[StackComponentType, str]
+    def _save_stack(
+        self,
+        name: str,
+        stack_configuration: Dict[StackComponentType, str],
+        is_update: bool = False,
     ) -> None:
-        """Add a stack to storage.
+        """Save a stack.
 
         Args:
             name: The name to save the stack as.
             stack_configuration: Dict[StackComponentType, str] to persist.
         """
         with Session(self.engine) as session:
-            stack = ZenStack(name=name, created_by=1)
-            session.add(stack)
-            for ctype, cname in stack_configuration.items():
-                if cname is not None:
-                    session.add(
-                        ZenStackDefinition(
-                            stack_name=name,
-                            component_type=ctype,
-                            component_name=cname,
+            if is_update:
+                # CHECK THAT THE STACK EXISTS FIRST
+                for ctype, cname in stack_configuration.items():
+                    if cname is not None:
+                        statement = (
+                            select(ZenStackDefinition)
+                            .where(ZenStackDefinition.stack_name == name)
+                            .where(ZenStackDefinition.component_type == ctype)
+                        )
+                        results = session.exec(statement)
+                        component = results.one()
+                        if component is None:
+                            session.add(
+                                ZenStackDefinition(
+                                    stack_name=name,
+                                    component_type=ctype,
+                                    component_name=cname,
+                                )
+                            )
+                        else:
+                            component.component_name = cname
+                            component.component_type = ctype
+                            session.add(component)
+                        session.commit()
```

Comment:
Not sure if this commit is necessary as it already happens right at the end of the method

Reply:
You're right. I think I was worried about the possibility of things getting out of sync, but with atomic commits to the database, this shouldn't make any difference. I'll remove this.